### PR TITLE
Addressed issue #1100 from Thimble. Made the sidebar switch color along when the theme changes

### PR DESCRIPTION
**Pull request by DorelBarbu**
In the sidebarTheme.css there are now to classes for the sidebar object, the 'dark' class, and the 'light' class. 

In the ThemeManager.js file, I modified the loadCurrentTheme function such that it toggles the appropriate class on the sidebar element and makes sure that there is always a 'dark' or 'light' class active, the default being the 'light' class.

### CODE REVIEW

**Comment by flukeout** [comment]
It seems like there is a bunch of new logic there related to toggling classes, which I think gets a bit confusing, what if we just used one attribute on the body like..

``<body data-theme='light-theme'>`` 

...then we could just easily set it once. We have access to the ``theme.name`` property. Then we can use that attribute in the CSS

body[data-theme='light-theme'] {
  /* styles * /
}

File: src/view/ThemeManager.js [hunk 1]
```diff
@@ -238,9 +238,12 @@ define(function (require, exports, module) {
                 })
                 .then(function (cssContent) {
                     $("body").toggleClass("dark", theme.dark);
+                    $("body").toggleClass("light");
                     styleNode.text(cssContent);
-
                     pending.resolve(theme);
+                    if($("body").hasClass("dark") === false && $("body").hasClass("light" ) === false){
```
Comment (Pomax) [inline src/view/ThemeManager.js]: If you want to test that `body` does not have "more than one class" it's actually easier to do that with a single selector:
```
if($("body.dark body.light").length() === 0) {
  // there is no 'dark', or 'light', class attached to the body element
}
```

Comment (flukeout) [inline src/view/ThemeManager.js]: I think we should move away from body classes anyway and try it with an attribute, much cleaner.

Comment (DorelBarbu) [inline src/view/ThemeManager.js]: Thank you, Pomax! I've just updated the approach as flukeout suggested. Let's see how it goes from here.

**Comment by flukeout** [comment]
Ok - testing now.

**Comment by flukeout** [comment]
Re-testing!

**Comment by DorelBarbu** [comment]
Waiting for the verdict. :)

**Comment by flukeout** [comment]
Okay, one final change and we'll be good to go. We need to adjust the [light-theme] style for ``a.context`` node, starting on line 161. When you right-click a menu item in the filetree, it gets the .context style, but we only have a dark-theme style for it. See below, it turns 'lighter' - I think we should just use the same treatment as when the file is being hovered.

hover

**Comment by DorelBarbu** [comment]
Yes, of course, I will handle it.

**Comment by humphd** [comment]
This is looking really good, nice work @DorelBarbu.

**Comment by DorelBarbu** [comment]
Thank you, @humphd! It will be even better after I address @flukeout 's review.

File: src/extensions/default/bramble/stylesheets/sidebarTheme.css [hunk 2]
```diff
@@ -161,7 +161,7 @@ li.jstree-open > ul:empty:after {
 li.jstree-closed a.context-node,
 li.jstree-open a.context-node,
 li.jstree-leaf a.context-node {
-    background: rgba(255,255,255,0.08);
+    background: rgba(255,255,255,.08);
```
Comment (DorelBarbu) [inline src/extensions/default/bramble/stylesheets/sidebarTheme.css]: This commit should be ignore. Sorry, guys..

**Comment by flukeout** [comment]
Ok awesome! Theme looks great + good work @DorelBarbu 👍 🖌 🎨

**Comment by DorelBarbu** [comment]
Thank you, @flukeout! Can't wait to work on something else

**Pull request merged**